### ISSUE TITLE
If i want to use "where" inside whereNotExists inside join it makes "on" clause instead.

### ISSUE THREAD

**Issue by dominik59** [laravel/framework#27304]
- Laravel Version: 5.3.0
- PHP Version: 7.1
- Database Driver & Version: MySQL InnoDB 

### Description:
If i want to use "where" inside whereNotExists inside join it makes "on" clause instead.

Assume that building the query looks like that:
```php
$foundCategoryPackages = $this->categoryPackage->select('package_category.*')
    ->join('packages', function (Builder $join) {
        $join->on('package_category.package_id', '=', 'packages.id')
            ->whereNotExists(function (Builder $query) {
                $query->select('package_category.id')
                    ->from('package_category')
                    ->join(
                        'package_items',
                        'package_category.id',
                        'package_items.category_id'
                    )->join(
                        'contract_items',
                        'package_items.id',
                        'contract_items.package_item_id'
                    )->where('package_category.package_id', 'packages.id');
            });
    })
    ->join('projects', 'packages.project_id', 'projects.id')
    ->where('projects.id', '4')
    ->get();
```

Then valid sql should look like that:

```sql
select `package_category`.*
from `package_category`
       inner join `packages` on `package_category`.`package_id` = `packages`.`id` and
                                     not exists(select `package_category`.`id`
                                                from `package_category`
                                                       inner join `package_items`
                                                                  on `package_category`.`id` =
                                                                     `package_items`.`category_package_id`
                                                       inner join `contract_items`
                                                                  on `package_items`.`id` =
                                                                     `contract_items`.`package_item_id`
                                                where `package_category`.`package_id` = packages.id)
       inner join `projects` on `packages`.`project_id` = `projects`.`id`
where `projects`.`id` = 4
```

Unfortunatelly it looks like that:
```sql
select `package_category`.*
from `package_category`
       inner join `packages` on `package_category`.`package_id` = `packages`.`id` and
                                     not exists(select `package_category`.`id`
                                                from `package_category`
                                                       inner join `package_items`
                                                                  on `package_category`.`id` =
                                                                     `package_items`.`category_package_id`
                                                       inner join `contract_items`
                                                                  on `package_items`.`id` =
                                                                     `contract_items`.`package_item_id`
                                                on `package_category`.`package_id` = packages.id)
       inner join `projects` on `packages`.`project_id` = `projects`.`id`
where `projects`.`id` = 4
```
("on" clause instead of "where" inside "not exists" subquery)

### Steps To Reproduce:
Make similar query, watch that there is "on" instead of "where".

**Comment by staudenmeir** [comment]
Laravel 5.3 is [not supported](https://laravel.com/docs/releases#support-policy) anymore. This has been fixed in Laravel 5.5.4 (#21008).

**Comment by driesvints** [comment]
Heya, unfortunately we don't support this version anymore. Please check out [our support policy](https://laravel.com/docs/releases) on which versions we are currently supporting. Can you please try to upgrade to the latest version and see if your problem persists? If so feel free to reply and we'll try to have a look.
